mantidproject/mantid
# Define D1B, D4 and D20 diffractometers at ILL

This PR adds the IDFs for D4C, D20, D1B diffractometers at ILL. 
They are generated with the scripts in that should be merged first.

* [mantidgeometry:PR83](https://github.com/mantidproject/mantidgeometry/pull/83)
* [mantidgeometry:PR84](https://github.com/mantidproject/mantidgeometry/pull/84)
* [mantidgeometry:PR85](https://github.com/mantidproject/mantidgeometry/pull/85)

Also for existing D2B, the L2 distance and monitor position are corrected.

Detailed detector definitions can be found here:
* [D1B](https://github.com/mantidproject/documents/blob/master/Project-Management/ILL/Instrument_Definitions/D1B/D1B.md)
* [D4](https://github.com/mantidproject/documents/blob/master/Project-Management/ILL/Instrument_Definitions/D4/D4.md)
* [D20](https://github.com/mantidproject/documents/blob/master/Project-Management/ILL/Instrument_Definitions/D20/D20.md)

The descriptions are preliminary confirmed by the instrument scientists.



**To test**:

`LoadEmptyInstrument` with the new IDFs, check the instrument view.
Make sure that the detector models match their descriptions, most importantly, pay attention on:

-  **2&theta; spacing** - the difference of thetas between two neighbouring pixels
-  **2&theta; coverage** - the full 2-theta acceptance of the detectors
-  **initial 2&theta;** - the angle of the first detector pixel.

Fixes #18452  .

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
